src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,270 @@
+   * If undefined, checkbox will not be shown.
+   * Recommended to use `Checkbox` component.
+   */
+  checkbox?: React.ReactNode;
+  /**
+   * Custom expander element. If `hasSubNodes` is false, it won't be shown.
+   */
+  expander?: React.ReactNode;
+  /**
+   * Content shown after `TreeNode`.
+   */
+  children?: React.ReactNode;
+} & Omit<CommonProps, 'id'>;
+
+/**
+ * `TreeNode` component to display node content within a `Tree`.
+ * Must be used inside `Tree` component to correctly set node `depth` and `subNodes`.
+ * @example
+  <TreeNode
+    nodeId={props.nodeId}
+    label={props.node.label}
+    sublabel={props.node.sublabel}
+    onExpanded={onExpanded}
+    onSelected={onSelectedNodeChange}
+    isDisabled={props.isDisabled}
+    isExpanded={props.isExpanded}
+    isSelected={props.isSelected}
+    checkbox={
+      <Checkbox variant='eyeball' disabled={props.isDisabled} />
+    }
+    icon={<SvgPlaceholder />}
+  />
+ */
+export const TreeNode = (props: TreeNodeProps) => {
+  const {
+    nodeId,
+    label,
+    sublabel,
+    children,
+    style,
+    className,
+    icon,
+    hasSubNodes = false,
+    isDisabled = false,
+    isExpanded = false,
+    isSelected = false,
+    onSelected,
+    onExpanded,
+    checkbox,
+    expander,
+    ...rest
+  } = props;
+  useTheme();
+
+  const context = React.useContext(TreeContext);
+  const nodeDepth = context?.nodeDepth ?? 0;
+  const subNodeIds = context?.subNodeIds ?? [];
+  const parentNodeId = context?.parentNodeId;
+  const groupSize = context?.groupSize ?? 0;
+  const indexInGroup = context?.indexInGroup ?? 0;
+
+  const [isFocused, setIsFocused] = React.useState(false);
+  const nodeRef = React.useRef<HTMLLIElement>(null);
+
+  const styleDepth = React.useMemo(
+    () =>
+      getWindow()?.CSS?.supports?.(`--level: ${nodeDepth}`)
+        ? { '--level': nodeDepth, ...style }
+        : { marginLeft: nodeDepth ? nodeDepth * 28 : 0, ...style },
+    [nodeDepth, style],
+  );
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLLIElement>) => {
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        if (isExpanded) {
+          onExpanded(nodeId, false);
+        } else if (parentNodeId) {
+          const parentNode = nodeRef.current?.ownerDocument.querySelector(
+            `#${parentNodeId}`,
+          ) as HTMLElement;
+          parentNode?.focus();
+        }
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        if (!isExpanded && hasSubNodes) {
+          onExpanded(nodeId, true);
+        } else if (subNodeIds.length > 0) {
+          const subNodes = Array.from(
+            nodeRef.current?.ownerDocument.querySelectorAll(
+              `${subNodeIds.map((id) => `#${id}`).join(',')}`,
+            ) || [],
+          ).filter(
+            (el) => el.getAttribute('aria-disabled') !== 'true',
+          ) as HTMLElement[];
+          subNodes[0]?.focus();
```

Comment:
The code for that is ugly because you need to get `onExpanded`, sub-nodes IDs. I don't think `getFocusableNodes` would be beneficial here as it will get all focusable elements when I need only sub-nodes.